File: src/main/java/ch/grassl/common/Importer.java
package ch.grassl.common;

import ch.grassl.util.ResourceReader;

import java.util.List;

public interface Importer<T> {

    List<T> importData(String resource);

    static String[] read(String resource) {
        return ResourceReader.of(resource).read();
    }
}

File: src/main/java/ch/grassl/common/Importer.kt
package ch.grassl.common

import ch.grassl.util.ResourceReader

interface Importer<T> {

    fun importData(resource: String): List<T>

    companion object {
        @JvmStatic
        fun read(resource: String): Array<String> {
            return ResourceReader.of(resource).read()
        }
    }
}